src/casl/casl-ability.factory.ts

```diff
@@ -54,40 +54,199 @@ export type AppAbility = MongoAbility<PossibleAbilities, Conditions>;
 
 @Injectable()
 export class CaslAbilityFactory {
-  createForUser(user: JWTUser) {
+  accessEndpointForUser(user: JWTUser, subject: string) {
+    switch (subject) {
+      case "datasets":
+        return this.accessDatasetEndpointForUser(user);
+      case "elastic-search":
+        return this.accessElasticSearchEndpointForUser(user);
+      case "jobs":
+        return this.accessJobsEnpointForUser(user);
+      case "instruments":
+        return this.accessInstrumentEndpointForUser(user);
+      case "logbooks":
+        return this.accessLogbookEndpointForUser(user);
+      case "origdatablocks":
+        return this.accessOrigDatablockEndpointForUser(user);
+      case "policies":
+        return this.accessPolicyEndpointForUser(user);
+      case "proposals":
+        return this.accessProposalsEndpointForUser(user);
+      case "publisheddata":
+        return this.accessPublishedDataEndpointForUser(user);
+      case "samples":
+        return this.accessSamplesEndpointForUser(user);
+      case "users":
+        return this.accessUserEndpointForUser(user);
+      case "useridentities":
+        return this.accessUserEndpointForUser(user);
```

Comment:
We can avoid duplication here by 
```suggestion
      case "users":
      case "useridentities":
        return this.accessUserEndpointForUser(user);
```